### ISSUE TITLE
Multiline if-else rule could also consider for and while-loop

### ISSUE THREAD

**Issue by hendraanggrian** [pinterest/ktlint#2223]
Forcing braces in multiline if-else improves readability, so why not extend it to all control flow statements: `for`, `while`, and `if`?

```kotlin
fun hasDigit(string: String): Boolean {
    for (character in string)
        if (character.isDigit())
            return true
    return false
}
```

## Expected Behavior

Violations:
- `for (character in string)`
- `if (character.isDigit())`

## Current Behavior

Violations:
- `if (character.isDigit())`

## Additional information

- Current version of ktlint: 0.49.1

**Comment by paul-dingemans** [comment]
[Kotlin Coding coventions](https://kotlinlang.org/docs/coding-conventions.html) do not mention braces for multiline while, do-while and for loops. 

[Android Kotlin style guide](https://developer.android.com/kotlin/style-guide#braces) is a bit more explicit:

> Braces are otherwise required for any if, for, when branch, do, and while statements and expressions, even when the body is empty or contains only a single statement.

I do think this behavior fits well for `ktlint_official` code style, and for `android_studio` code style because it is mentioned explicitly.

This behavior is to be added to new experimental rule `control-flow-braces`. Also note that the block of a multiline if-statement without else should be wrapped between braces.

**Comment by paul-dingemans** [comment]
Resolved by #2298
